src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt

```diff
@@ -246,6 +246,84 @@ internal class ActivitiesServiceTest {
     }
   }
 
+  @Nested
+  inner class Deallocate {
+
+    @Test
+    fun `should log an allocation event`() {
+
+      whenever(activitiesApiService.getAllocation(allocationId)).thenReturn(newAllocation())
+      whenever(mappingService.getMappingGivenActivityScheduleId(activityScheduleId)).thenReturn(
+        ActivityMappingDto(
+          nomisCourseActivityId = nomisCourseActivityId,
+          activityScheduleId = activityScheduleId,
+          mappingType = "ACTIVITY_CREATED",
+        )
+      )
+      whenever(nomisApiService.deallocate(eq(nomisCourseActivityId), any(), any())).thenReturn(
+        OffenderProgramProfileResponse(offenderProgramReferenceId = offenderProgramReferenceId)
+      )
+
+      activitiesService.deallocate(
+        AllocationDomainEvent(
+          eventType = "dummy",
+          scheduleId = activityScheduleId,
+          allocationId = allocationId,
+          version = "1.0",
+          description = "description",
+          occurredAt = LocalDateTime.now(),
+        )
```

Comment:
I guess it doesn't matter as the events haven't been defined by the producer yet, but this doesn't fit the domain event schema - scheduleId and allocationId should be in an `additionalInformation` section and there should be a detail URL that points back to the Activities API.

I'd be slightly worried that the Activities team just copy this structure when they get around to producing events.

Reply:
Done